src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java

```diff
@@ -52,4 +60,161 @@ protected KualiDecimal getPercentage(KualiDecimal amount, BigDecimal percent) {
+            e.setFinancialObjectCode(et.getObjectCode());
+            e.setFinancialSubObjectCode(et.getSubObjectCode());
+        } else {
+            e.setFinancialObjectCode(icrRateDetail.getFinancialObjectCode());
+            e.setFinancialSubObjectCode(icrRateDetail.getFinancialSubObjectCode());
+        }
+
+        if (GeneralLedgerConstants.PosterService.SYMBOL_USE_EXPENDITURE_ENTRY.equals(icrRateDetail.getAccountNumber())) {
+            e.setAccountNumber(et.getAccountNumber());
+            e.setChartOfAccountsCode(et.getChartOfAccountsCode());
+            e.setSubAccountNumber(et.getSubAccountNumber());
+        } else if (GeneralLedgerConstants.PosterService.SYMBOL_USE_ICR_FROM_ACCOUNT.equals(icrRateDetail.getAccountNumber())) {
+            e.setAccountNumber(icrGenerationMetadata.getIndirectCostRecoveryAcctNbr());
+            e.setChartOfAccountsCode(icrGenerationMetadata.getIndirectCostRcvyFinCoaCode());
+            e.setSubAccountNumber(KFSConstants.getDashSubAccountNumber());
+        } else {
+            e.setAccountNumber(icrRateDetail.getAccountNumber());
+            e.setSubAccountNumber(icrRateDetail.getSubAccountNumber());
+            e.setChartOfAccountsCode(icrRateDetail.getChartOfAccountsCode());
+            // TODO Reporting thing line 1946
+        }
+        // take care of infinite recursive error case - do not generate entries
+        if ((et.getAccountNumber().equals(e.getAccountNumber())) &&
+            (et.getChartOfAccountsCode().equals(e.getChartOfAccountsCode())) &&
+            (et.getSubAccountNumber().equals(e.getSubAccountNumber())) &&
+            (et.getObjectCode().equals(e.getFinancialObjectCode())) &&
+            (et.getSubObjectCode().equals(e.getFinancialSubObjectCode()))) {
+            List<Message> warnings = new ArrayList<Message>();
+            warnings.add(new Message("Infinite recursive encumbrance error " + et.getChartOfAccountsCode() + " " + et.getAccountNumber() + " " + et.getSubAccountNumber() + " " + et.getObjectCode() + " " + et.getSubObjectCode(), Message.TYPE_WARNING));
+            reportWriterService.writeError(et, warnings);
+            return;
+        }
+
+        e.setFinancialDocumentTypeCode(parameterService.getParameterValueAsString(PosterIndirectCostRecoveryEntriesStep.class, KFSConstants.SystemGroupParameterNames.GL_INDIRECT_COST_RECOVERY));
+        e.setFinancialSystemOriginationCode(parameterService.getParameterValueAsString(KfsParameterConstants.GENERAL_LEDGER_BATCH.class, KFSConstants.SystemGroupParameterNames.GL_ORIGINATION_CODE));
+        SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT_STRING);
+        /*CUMod - start*/
+        StringBuffer docNbr = new StringBuffer(ICR_EDOC_PREFIX);
+        docNbr.append(sdf.format(runDate));
+        e.setDocumentNumber(docNbr.toString());
+        LOG.debug("CuPosterServiceImpl.generateTransactions: setDocumentNumber=" + docNbr.toString() + "=");
```

Comment:
Do you need the extra "=" sign at the end of the log text? If not, then I would suggest removing it.

Reply:
My intent of the extra equal sign is to find null or blank Strings that we sometimes get as bugs.  Those should not occur here but I wold prefer to leave it in unless you are really opposed to it.

Reply:
What about using single or double quotes around the printed doc number? I think that would make the formatting clearer, and would make it easier to spot empty or blank values.

Reply:
I'll change the "=" to single quotes.